app/src/main/java/org/mozilla/focus/web/WebViewProvider.kt

```diff
@@ -0,0 +1,83 @@
+package org.mozilla.focus.web
+
+import android.content.Context
+import android.preference.PreferenceManager
+import android.util.AttributeSet
+import android.view.View
+import android.webkit.WebSettings
+import org.mozilla.focus.web.Renderers.WEBVIEW
+import org.mozilla.focus.web.Renderers.GECKO
+import org.mozilla.focus.webview.SystemWebView
+
+const val ENGINE_PREF_STRING_KEY = "engine_choice"
+
+object WebViewProvider {
+    var engineName: String = Config.renderer.name
+    var engine : IWebViewProvider? = null
+
+    fun writeEnginePref(context: Context, newEngineName: String = engineName) {
+        requireNotNull(context)
+        val prefs = PreferenceManager.getDefaultSharedPreferences(context)
+        prefs.edit().putString(ENGINE_PREF_STRING_KEY, newEngineName).apply()
+    }
+
+    fun readEnginePref(context: Context) {
+        requireNotNull(context)
```

Comment:
Can you help me understand requireNotNull here? Won't the Kotlin compiler add a null check already because Context is not nullable?

Reply:
I was checking preconditions from the Java callers. It's quite possible this is redundant. I can check the bytecode.

Reply:
Checked the bytecode. It was redundant, but the compiler is smart enough to check only once. I'l remove the extra checks.